### PR TITLE
Tao 5780/wk1009 unable to add delivery in the test center

### PR DESCRIPTION
**Pull request by uncleempty**
Added resize func to resize eligibility editor modal. now it will not run under the bottom browser edge.

### CODE REVIEW

File: views/js/component/eligibilityEditor.js [hunk 1]
```diff
@@ -254,7 +255,20 @@ define([
...
+
+            /**
+             * Changes eligibilityEditor modal height and adds scroll for the modal to fit the screen height
+             *
+             * @returns {eligibilityEditor} chains the component
+             */
+            resize: function resize() {
+                eligibilityEditor.getElement().css('top', '40px');
+                eligibilityEditor.getElement().css('max-height', ($(window).height() - 80) + 'px');
+                eligibilityEditor.getElement().css('overflow-y', 'scroll');
```
Comment (jsconan) [inline views/js/component/eligibilityEditor.js]: It should be better to gather all of this CSS changes into one single call, unless each command is dependent to the previous:
```javascript
eligibilityEditor.getElement().css({
    'top': '40px',
    'max-height': ($(window).height() - 80) + 'px',
    'overflow-y': 'scroll'
});
```

File: views/js/component/eligibilityEditor.js [hunk 2]
```diff
@@ -254,7 +255,20 @@ define([
...
+            },
+
+            /**
+             * Changes eligibilityEditor modal height and adds scroll for the modal to fit the screen height
+             *
+             * @returns {eligibilityEditor} chains the component
+             */
+            resize: function resize() {
+                eligibilityEditor.getElement().css('top', '40px');
+                eligibilityEditor.getElement().css('max-height', ($(window).height() - 80) + 'px');
```
Comment (jsconan) [inline views/js/component/eligibilityEditor.js]: Where do those values come from? If I understand well 40 is the top margin, and 80 is then the sum of the top and bottom margins. But why those values, and why is it hardcoded?

**Review by jsconan — CHANGES_REQUESTED** [review]
The companion PR on taoCore does not seem to be used here. Moreover I still have the issue on my browser: when I shrink the window to have a small height, the modal is still moved outside...

File: views/js/component/eligibilityEditor.js (hunk 1 above)
Comment (uncleempty) [reply views/js/component/eligibilityEditor.js]: Agree, will move it.

File: views/js/component/eligibilityEditor.js (hunk 2 above)
Comment (uncleempty) [reply views/js/component/eligibilityEditor.js]: Okay. It's simple: we cannot guarantee that inital top margin (which is 40px) isn't changed and calculated into another value till this moment. Then, if we are moving all those 'setters' to single call - the actual top margin which will be suitable won't be accessible.
Possibly, this variant would be a little better:
```
eligibilityEditor.getElement().css({
    'top': '40px',
    'max-height': ($(window).height() - 80) + 'px',
    'overflow-y': 'scroll'
});
eligibilityEditor.getElement().css('max-height', ($(window).height() - (parseInt(eligibilityEditor.getElement().css('top'),10)*2)) + 'px');
```

**Comment by uncleempty** [comment]
Answer for the last comment:
Yes, resizing is not made "on-the-fly", it's made before modal is opened. If needed - I'll add handler for `$(window).resize` (or smth similar, don't remember which event stands for this) to resize modal automatically on window resize.

**Comment by jsconan** [comment]
Please avoid to use the `resize` event, unless there is a very strong need and you throttle it. This will lead to performances issues.

File: views/js/component/eligibilityEditor.js (hunk 2 above)
Comment (jsconan) [reply views/js/component/eligibilityEditor.js]: Well ok, the values should be fine if you add a comment to explain why.

**Comment by jsconan** [comment]
Let me explain better the issue I got:
- I shrink the window to have a lesser height
- then I try to add a delivery
- the modal is stil moved outside

This is not a resize issue, just the target issue is not fixed as I can still reproduce it.
So no need to listen to resize or something like that.

**Review by jsconan — APPROVED** [review]
Works fine! good job!

**Pull request merged**